api/routers/products.py

```
@@ -0,0 +1,97 @@
+from fastapi.responses import StreamingResponse
+
+from crud import product_crud
+from core import schemas
+from services import product_service
+from api.dependencies import get_db, get_current_user
+
+router = APIRouter()
+logger = logging.getLogger(__name__)
+
+@router.post("/upload-products", tags=["Products"], dependencies=[Depends(get_current_user)])
+async def upload_products(tenant_id: str, file: UploadFile, db: Session = Depends(get_db)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)